src/basic/parse-util.c

```diff
@@ -589,3 +589,13 @@ int parse_ip_port(const char *s, uint16_t *ret) {
 
         return 0;
 }
+
+int parse_dev(const char *s, dev_t *ret) {
+        unsigned major, minor;
+
+        if (sscanf(s, "%u:%u", &major, &minor) != 2)
+                return -EINVAL;
+
+        *ret = makedev(major, minor);
```

Comment:
while we are at it, I think it would be nice to be stricter here on the range of the two parts we accept. Unfortunately libc doesn't expose information about the valid range, but an OK check could be to immediately split major/minor out again, and compare it with the original values, to ensure the full information is retained and doesn't get mangled up. i.e., something like this:

```c
…
unsigned x, y;
dev_t d;
…
if (sscanf(s, "%u:%u", &x, &y) != 2)
        return -EINVAL;

d = makedev(x, y);
if ((unsigned) major(d) != x || (unsigned) minor(d) != y)
        return -EINVAL;

*ret = d;
…
```

i.e. i want that parse_dev("4294967295:4294967295", &ret) results in a parse failure. With your current code this would parse successfully, but ret would contain nonsense data.